distributed/tests/test_core.py

```diff
@@ -640,22 +633,17 @@ async def connect_to_server():
 
     tasks = [asyncio.create_task(connect_to_server()) for _ in range(30)]
 
-    await asyncio.sleep(0)
-    assert pool._connecting
-    close_fut = asyncio.create_task(pool.close())
+    while not pool._connecting:
+        await asyncio.sleep(0.01)
 
-    with pytest.raises(
-        CommClosedError, match="ConnectionPool not running. Status: Status.closed"
-    ):
-        await asyncio.gather(*tasks)
+    close_fut = asyncio.create_task(pool.close())
 
+    done, _ = await asyncio.wait(tasks)
     await close_fut
+    assert all(t.cancelled() for t in done)
```

Comment:
do you need the close_fut anymore?

```suggestion
    await pool.close()
    done, _ = await asyncio.wait(tasks)
    assert all(t.cancelled() for t in done)
```